test/11_dependency_versions/cibuildwheel_test.py

```diff
@@ -0,0 +1,105 @@
+import os
+import re
+import pytest
+import textwrap
+import cibuildwheel.util
+
+import utils
+
+
+def get_version_from_constraint_file(package_name, constraint_file):
+    version_pattern = package_name + r'==([^\s]+)'
+
+    with open(constraint_file, encoding='utf8') as f:
+        constraint_file_text = f.read()
+
+    match = re.search(version_pattern, constraint_file_text)
+
+    if not match:
+        raise Exception("couldn't find version spec matching " + version_pattern)
+
+    return match.group(1)
```

Comment:
Maybe instead of repetitive opening files use `re.findall` and return dict?

Reply:
Should be possible as well. Though it's a test, so I'd say clarity (and certainty of correctness) is more important than efficiency here.

Reply:
Yes, it's nicer using re.findall. thanks!